tests/monitoring_test.go

```diff
@@ -0,0 +1,534 @@
+			promRule := getPrometheusRule(f)
+			for _, group := range promRule.Spec.Groups {
+				if group.Name == "cdi.rules" {
+					for _, rule := range group.Rules {
+						if rule.Alert != "" {
+							Expect(rule.Labels).ToNot(BeNil())
+							checkForSeverityLabel(rule)
+							checkForHealthImpactLabel(rule)
+							checkForPartOfLabel(rule)
+							checkForComponentLabel(rule)
+						}
+					}
+				}
+			}
+		})
+	})
+
+	Context("[rfe_id:7101][crit:medium][vendor:cnv-qe@redhat.com][level:component] Alert tests", func() {
+		var originalPromRule *promv1.PrometheusRule
+
+		BeforeEach(func() {
+			originalPromRule = getPrometheusRule(f)
+		})
+
+		AfterEach(func() {
+			setPrometheusRule(f, originalPromRule)
+		})
+
+		It("[test_id:7962] CDIOperatorDown alert firing when operator scaled down", func() {
+			deploymentName := "cdi-operator"
+			By("Scale down operator so alert will trigger")
+			originalReplicas := scaleDeployment(f, deploymentName, 0)
+			Eventually(func() bool {
+				dep, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return dep.Status.Replicas == 0
+			}, 20*time.Second, 1*time.Second).Should(BeTrue())
+
+			setPrometheusRuleAlertDuration(f, originalPromRule, "CDIOperatorDown", "0m")
+			waitForPrometheusAlert(f, "CDIOperatorDown")
+
+			By("Ensuring original value of replicas restored")
+			scaleDeployment(f, deploymentName, originalReplicas)
+			err := utils.WaitForDeploymentReplicasReady(f.K8sClient, f.CdiInstallNs, deploymentName)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should throw CDIDataVolumeUnusualRestartCount alert when a DataVolume object restarts more than three times", func() {
+			By("creating a DataVolume")
+			dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, "500Mi", nonExistentImageURL)
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+
+			By("waiting for the DataVolume to be restarted more than three times")
+			Eventually(func() int32 {
+				dv, err := f.CdiClient.CdiV1beta1().DataVolumes(f.Namespace.Name).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return dv.Status.RestartCount
+			}, 5*time.Minute, 10*time.Second).Should(BeNumerically(">", 3))
+
+			By("checking that the CDIDataVolumeUnusualRestartCount alert is firing")
+			setPrometheusRuleAlertDuration(f, originalPromRule, "CDIDataVolumeUnusualRestartCount", "0m")
+			waitForPrometheusAlert(f, "CDIDataVolumeUnusualRestartCount")
+
+			By("deleting the DataVolume")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should throw CDIStorageProfilesIncomplete alert if there is a incomplete StorageProfile", func() {
```

Comment:
Same point about duplicating tests, we can just add the alert check to the existing test,
as these are pretty pricey tests runtime wise